lib/ansible/modules/cloud/amazon/iam_user_info.py

```diff
@@ -0,0 +1,187 @@
+     - If specified, then would get all user names whose path starts with user provided value.
+    required: false
+    default: '/'
+    type: str
+requirements:
+  - botocore
+  - boto3
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+# Gather facts about "test" user.
+- name: Get IAM user facts
+  iam_user_info:
+    name: "test"
+
+# Gather facts about all users in the "dev" group.
+- name: Get IAM user facts
+  iam_user_info:
+    group: "dev"
+
+# Gather facts about all users with "/division_abc/subdivision_xyz/" path.
+- name: Get IAM user facts
+  iam_user_info:
+    path: "/division_abc/subdivision_xyz/"
+'''
+
+RETURN = r'''
+iam_users:
+    description: list of maching iam users
+    returned: success
+    type: complex
+    contains:
+        arn:
+            description: the ARN of the user
+            returned: if user exists
+            type: str
+            sample: "arn:aws:iam::156360693172:user/dev/test_user"
+        create_date:
+            description: the datetime user was created
+            returned: if user exists
+            type: str
+            sample: "2016-05-24T12:24:59+00:00"
+        password_last_used:
+            description: the last datetime the password was used by user
+            returned: if password was used at least once
+            type: str
+            sample: "2016-05-25T13:39:11+00:00"
+        path:
+            description: the path to user
+            returned: if user exists
+            type: str
+            sample: "/dev/"
+        user_id:
+            description: the unique user id
+            returned: if user exists
+            type: str
+            sample: "AIDUIOOCQKTUGI6QJLGH2"
+        user_name:
+            description: the user name
+            returned: if user exists
+            type: str
+            sample: "test_user"
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info, AWSRetry
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError, ParamValidationError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+@AWSRetry.exponential_backoff()
+def list_iam_users_with_backoff(client, operation, **kwargs):
+    paginator = client.get_paginator(operation)
+    return paginator.paginate(**kwargs).build_full_result()
+
+
+def list_iam_users(connection, module):
+
+    name = module.params.get('name')
+    group = module.params.get('group')
+    path = module.params.get('path')
+
+    params = dict()
+    iam_users = []
+
+    if not group and not path:
+        if name:
+            params['UserName'] = name
+        try:
+            iam_users.append(connection.get_user(**params)['User'])
+        except (ClientError, ParamValidationError) as e:
```

Comment:
any reason this doesn't use BotoCoreError rather than ParamValidationError?

I'm a bit worried this might not catch ProfileNotFound or permissions errors (one of the two)

Reply:
Nice catch, I'd assumed they were ClientErrors, just checked, they're BotoCoreErrors

Reply:
https://github.com/ansible/ansible/pull/23382/commits/fdeb8d2f39154ab8e6d072a349d69726a7b6e9b8 switches this.

Reply:
https://github.com/ansible/ansible/pull/23382/commits/a1dcf29f24549a6879e90ca9b8b1e7c98231718f adds some tests for this as well.